__tests__/transform/object.test.js

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
+ * the License. A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+var transformObject = require('../../lib/transform/object');
+
+var options = {
+  transforms: [
+    {
+      type: 'attribute',
+      transformer: function() {
+        return {foo: 'bar'}
+      }
+    }, {
+      type: 'attribute',
+      transformer: function() {
+        return {bar: 'foo'}
+      }
+    }, {
+      type: 'name',
+      matcher: function(prop) {
+        return prop.attributes.foo === 'bar';
+      },
+      transformer: function() {
+        return "hello";
+      }
+    }
+  ]
+};
+
+describe('transform', () => {
+  describe('object', () => {
+    it('does not crash when called without parameters', () => {
+      expect(transformObject()).toEqual({});
+    })
+
+    it('returns expected object', () => {
```

Comment:
what do you think of splitting this test in smaller tests to check the different cases that I see in the code for `lib/transform/object` ("is plain object and has value", "is just plain object", "else")? no strong opinions on this, just an idea

Reply:
I think smaller tests are better, they will be easier to debug if a test breaks. Really each test should test 1 thing.